deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml

```diff
@@ -30,6 +30,18 @@ spec:
 {{ toYaml .Values.otellogs.daemonset.podLabels | indent 8 }}
 {{- end }}
     spec:
+{{- if .Values.otellogs.daemonset.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.otellogs.daemonset.nodeSelector | indent 8 }}
+{{- end }}
+{{- if or .Values.otellogs.daemonset.affinity }}
+      affinity:
+{{ toYaml .Values.otellogs.daemonset.affinity | indent 8 }}
+{{- end }}
+{{- if .Values.otellogs.daemonset.tolerations }}
+      tolerations:
+{{ toYaml .Values.otellogs.daemonset.tolerations | indent 8 }}
+{{- end }}
```

Comment:
Those `if` blocks are nice to have but since you've added the default in `values.yaml` I'm wondering if we should keep them.

Reply:
IMHO, generated template is nicer with ifs than:
```
nodeSelector:
  {}
```

Please create gh issue for that if you want to change it, as all other templates should be reviewed and changed as well